src/main/scala/uk/gov/nationalarchives/Processor.scala

```diff
@@ -194,8 +204,9 @@ object Processor {
       config: Config,
       sqsClient: DASQSClient[IO],
       ocflService: OcflService,
-      entityClient: EntityClient[IO, Fs2Streams[IO]]
+      entityClient: EntityClient[IO, Fs2Streams[IO]],
+      xmlValidator: ValidateXmlAgainstXsd[IO]
```

Comment:
You don't really need this parameter in the apply method. You could have it as 
```scala
def apply(
      config: Config,
      sqsClient: DASQSClient[IO],
      ocflService: OcflService,
      entityClient: EntityClient[IO, Fs2Streams[IO]]
  ): IO[Processor] = IO(
    new Processor(config, sqsClient, ocflService, entityClient, ValidateXmlAgainstXsd[IO](XipXsdSchemaV6))
  )
```
That way, you don't need to change anything in the Main class and your tests will still work because you're calling `new Processor` in them. I can't see us needing to validate anything other than the Xip schema. What do you think?

Reply:
Ah, yes, that sounds better